### PR TITLE
Implement Metrics for coala: Improve Description

### PR DESCRIPTION
**Pull request by dracarys09**
**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

File: data/projects/implement_metrics_for_coala.md [hunk 1]
```diff
@@ -7,28 +7,46 @@ them can be measured objectively by so-called
...
 calculate these metrics from a piece of code.
 
+coala should have a module that can produce software metrics because software
+metrics are crucial when it comes to quality assurance. Source code metrics
+for quality assurance aims at writing code with less bugs inside and writing
+code for easy maintenance. Both lead to writing code as simple as possible
+with few elements in each unit. Picking right metrics is paramount and this
+project aims at implementing an API and a set of Bears for producing
+software metrics.
+
```
Comment (sims1253) [inline data/projects/implement_metrics_for_coala.md]: I like what you added but feel like the first three paragraphs could be combined for more fluent reading. Some stuff is in there multiple times

File: data/projects/implement_metrics_for_coala.md [hunk 2]
```diff
@@ -7,28 +7,46 @@ them can be measured objectively by so-called
...
 has to include multiple bears that calculate metrics for different
 programming languages.
 
+
 #### Milestones
 
-##### GSOC 2017 COMMUNITY BONDING
+##### PREPARATION/BONDING
 
+* Research existing tools for software metrics.
```
Comment (sims1253) [inline data/projects/implement_metrics_for_coala.md]: Make milestones testable and rather say "A list of software metric tools to support exists" because this is a state that can easily be checked against

File: data/projects/implement_metrics_for_coala.md [hunk 3]
```diff
@@ -7,28 +7,46 @@ them can be measured objectively by so-called
...
-* The API is properly documented and fully tested.
+* The API for bears to yield metrics is done and well documented.
 * At least one Bear implementing the API exists.
 * A working (not clean or merged!) prototype of the CLI visualization
 exists.
 
-##### GSOC 2017 FINAL
+##### CODING PHASE 2
+
+* The API for bears to yield metrics is merged into coala.
```
Comment (sims1253) [inline data/projects/implement_metrics_for_coala.md]: this is kind of the same as the first one. I'd put the completion of api concept in the preparation, implementation in first phase and then 1 bear in first phase, UI stuff in second phase more bears and tutorial 3rd phase. All code has to be tested and documented when it's merged so the "testing has begun" milestone is wrong as it has to be done together with the implementation

File: data/projects/implement_metrics_for_coala.md [hunk 4]
```diff
@@ -7,28 +7,40 @@ them can be measured objectively by so-called
 offers a good base for the implementation or wrapping of algorithms that
 calculate these metrics from a piece of code.
 
-The goal of this project proposal should be to implement handling and
-display of software metrics in coala. To prove the concept, this project
-has to include multiple bears that calculate metrics for different
-programming languages.
+coala should have a module that can produce software metrics becausethey are
```
Comment (RohanVB) [inline data/projects/implement_metrics_for_coala.md]: "becausethey"->"because they"

File: data/projects/implement_metrics_for_coala.md [hunk 5]
```diff
@@ -7,28 +7,40 @@ them can be measured objectively by so-called
...
 
-##### GSOC 2017 FINAL
+##### CODING PHASE 2
 
 * The coala CLI can visualize metrics.
 * The other coala binaries can handle metrics.
-* User documentation is written.
+* The API and Bears are fully documented and tested.
+
+
```
Comment (RohanVB) [inline data/projects/implement_metrics_for_coala.md]: extra newline?

File: data/projects/implement_metrics_for_coala.md [hunk 6]
```diff
@@ -7,28 +7,40 @@ them can be measured objectively by so-called
...
+##### CODING PHASE 3
+
+* At least one more Bear implementing the API exists.
+* The Bears are fully documented and tested.
+* A tutorial is merged which contains instructions on how to generate and
+visualize metrics for a project.
```
Comment (RohanVB) [inline data/projects/implement_metrics_for_coala.md]: double space this to make it inline with the above sentence

File: data/projects/implement_metrics_for_coala.md [hunk 7]
```diff
@@ -7,28 +7,40 @@ them can be measured objectively by so-called
...
-display of software metrics in coala. To prove the concept, this project
-has to include multiple bears that calculate metrics for different
-programming languages.
+coala should have a module that can produce software metrics becausethey are
+crucial when it comes to quality assurance. Source code metrics
+for quality assurance aims at writing maintainable code with less bugs.
+Picking right metrics is paramount and this project aims at implementing an
+API along with a set of Bears for handling and display of software metrics
+in coala. To prove the concept, a set of Bears implemented as a part of this
+project will calculate metrics for different programming languages.
```
Comment (RohanVB) [inline data/projects/implement_metrics_for_coala.md]: You're trying to prove the concept of the metrics project, but then you're saying that the proof will be a part of this project?

File: data/projects/implement_metrics_for_coala.md [hunk 8]
```diff
@@ -7,28 +7,41 @@ them can be measured objectively by so-called
...
-has to include multiple bears that calculate metrics for different
-programming languages.
+The goal of this project is to implement a module in coala that can produce
+software metrics because they are crucial when it comes to quality assurance.
+Source code metrics for quality assurance aims at writing maintainable code
+with less bugs. Picking right metrics is paramount and this project aims at
+implementing an API along with a set of Bears for handling and display of
+software metrics in coala. To prove the concept, a set of Bears implemented
+as a part of this project will calculate metrics for different programming
+languages.
```
Comment (RohanVB) [inline data/projects/implement_metrics_for_coala.md]: You're trying to prove the concept of the metrics project, but then you're saying that the proof will be a part of this project? Maybe reword the last sentence elaborate further? I'm not sure that if it's too much info for the project description, just my opinion :)

File: data/projects/implement_metrics_for_coala.md [hunk 9]
```diff
@@ -7,28 +7,41 @@ them can be measured objectively by so-called
...
 
-##### GSOC 2017 MIDTERM
+##### CODING PHASE 1
 
-* The API for bears to yield metrics has been implemented.
-* The API is properly documented and fully tested.
+* The API for bears to yield metrics is implemented.
 * At least one Bear implementing the API exists.
 * A working (not clean or merged!) prototype of the CLI visualization
 exists.
```
Comment (RohanVB) [inline data/projects/implement_metrics_for_coala.md]: double space this to make it inline with the above sentence

Comment (dracarys09) [inline data/projects/implement_metrics_for_coala.md]: @RohanVB done

File: data/projects/implement_metrics_for_coala.md (hunk 6 above)
Comment (dracarys09) [inline data/projects/implement_metrics_for_coala.md]: @RohanVB done

File: data/projects/implement_metrics_for_coala.md (hunk 5 above)
Comment (dracarys09) [inline data/projects/implement_metrics_for_coala.md]: done

File: data/projects/implement_metrics_for_coala.md (hunk 4 above)
Comment (dracarys09) [inline data/projects/implement_metrics_for_coala.md]: done

File: data/projects/implement_metrics_for_coala.md (hunk 8 above)
Comment (dracarys09) [inline data/projects/implement_metrics_for_coala.md]: This was already there in the description before...  :worried:

File: data/projects/implement_metrics_for_coala.md [hunk 10]
```diff
@@ -7,28 +7,40 @@ them can be measured objectively by so-called
...
+as a part of this project will calculate metrics for different programming
+languages.
+
 
 #### Milestones
 
-##### GSOC 2017 COMMUNITY BONDING
+##### PREPARATION/BONDING
 
+* A list of software metric tools to support exists.
```
Comment (RohanVB) [inline data/projects/implement_metrics_for_coala.md]: to support..? Looks incomplete.

Comment (dracarys09) [inline data/projects/implement_metrics_for_coala.md]: yeah I was having the same thought... but @sims1253 suggested to write this instead of "Research some existing tools for calculating software metrics"

File: data/projects/implement_metrics_for_coala.md (hunk 8 above)
Comment (RohanVB) [inline data/projects/implement_metrics_for_coala.md]: I don't think the "To prove the concept," line is necessary, if the approach is defined by having a set of bears to calculate metrics for respective languages, then we can just start with "A set of Bears..". What do you think about that? :D

Comment (dracarys09) [inline data/projects/implement_metrics_for_coala.md]: This makes more sense to me...

File: data/projects/implement_metrics_for_coala.md [hunk 11]
```diff
@@ -7,28 +7,40 @@ them can be measured objectively by so-called
...
 calculate these metrics from a piece of code.
 
-The goal of this project proposal should be to implement handling and
-display of software metrics in coala. To prove the concept, this project
-has to include multiple bears that calculate metrics for different
-programming languages.
+The goal of this project is to implement a module in coala that can produce
+software metrics because they are crucial when it comes to quality assurance.
+Source code metrics for quality assurance aims at writing maintainable code
+with less bugs. Picking right metrics is paramount and this project aims at
```
Comment (sims1253) [inline data/projects/implement_metrics_for_coala.md]: picking >>the<< right metrics

Comment (sims1253) [inline data/projects/implement_metrics_for_coala.md]: paramount -> important or hard, depending on what you want to say :D

File: data/projects/implement_metrics_for_coala.md (hunk 8 above)
Comment (sims1253) [inline data/projects/implement_metrics_for_coala.md]: Instead of proving the concept I'd say the bears exist as a showcase, orientation and maybe tutorial

Also please address my "combinet he first three paragraphs" comment from last iteration.
Make it so that there is an intro paragraph with a Motivation (why do we want/need this project) and a goal and a second paragraph with technical information.

File: data/projects/implement_metrics_for_coala.md [hunk 12]
```diff
@@ -1,34 +1,43 @@
-There is more to software quality than just passing builds -
-coala should support generating metrics for your code.
+There is more to software quality than just passing builds. coala should
```
Comment (sims1253) [inline data/projects/implement_metrics_for_coala.md]: Could you make a transition from the first to the second sentence? This doesn't sound nice.

Comment (sims1253) [inline data/projects/implement_metrics_for_coala.md]: Also you say coala should support metrics before talking about metrics and why coala should support them

Comment (dracarys09) [inline data/projects/implement_metrics_for_coala.md]: @sims1253 done...  :smile:

File: data/projects/implement_metrics_for_coala.md (hunk 11 above)
Comment (dracarys09) [inline data/projects/implement_metrics_for_coala.md]: done

File: data/projects/implement_metrics_for_coala.md (hunk 10 above)
Comment (SanketDG) [inline data/projects/implement_metrics_for_coala.md]: @sims1253 lines ending with "exists" looks ambiguous.

Can we have "Tools for calculating software metrics should be researched"

File: data/projects/implement_metrics_for_coala.md [hunk 13]
```diff
@@ -1,34 +1,46 @@
...
-exists.
+  exists.
 
-##### GSOC 2017 FINAL
+##### CODING PHASE 2
 
 * The coala CLI can visualize metrics.
 * The other coala binaries can handle metrics.
-* User documentation is written.
+* The API and Bears are fully documented and tested.
```
Comment (SanketDG) [inline data/projects/implement_metrics_for_coala.md]: No bears are written until now as per your phases. Maybe only document and test the API?

Comment (SanketDG) [inline data/projects/implement_metrics_for_coala.md]: uh nvm

File: data/projects/implement_metrics_for_coala.md [hunk 14]
```diff
@@ -1,34 +1,46 @@
...
 * The applicant has created CLI and web UI mockups for metrics.
 * The applicant has created a [cEPs](https://coala.io/cep) for his project.
 
-##### GSOC 2017 MIDTERM
+##### CODING PHASE 1
 
-* The API for bears to yield metrics has been implemented.
-* The API is properly documented and fully tested.
+* The API for bears to yield metrics is implemented.
 * At least one Bear implementing the API exists.
```
Comment (SanketDG) [inline data/projects/implement_metrics_for_coala.md]: so there is already one bear, or is this the state that is to be reached?

Comment (dracarys09) [inline data/projects/implement_metrics_for_coala.md]: @SanketDG  this state is to be reached...

Should I change it to "At least one Bear is implemented which uses this API" ??

Comment (SanketDG) [inline data/projects/implement_metrics_for_coala.md]: yes better

File: data/projects/implement_metrics_for_coala.md (hunk 13 above)
Comment (dracarys09) [inline data/projects/implement_metrics_for_coala.md]: I didn't quite understand... is it unclear from the sentence that the milestone requires that API and all the Bears are fully documented and tested ?

File: data/projects/implement_metrics_for_coala.md (hunk 10 above)
Comment (sims1253) [inline data/projects/implement_metrics_for_coala.md]: Milestones should be states that can be checked against so should be research is not really something you can easily test.

Comment (dracarys09) [inline data/projects/implement_metrics_for_coala.md]: ah... I addressed @SanketDG 's comment in my commit...

File: data/projects/implement_metrics_for_coala.md [hunk 15]
```diff
@@ -1,34 +1,46 @@
...
+Picking the right metrics is important and this project aims at implementing
+an API along with a set of Bears for handling and display of software metrics
+in coala.
 
 #### Milestones
 
-##### GSOC 2017 COMMUNITY BONDING
+##### PREPARATION/BONDING
 
+* Tools for calculating software metrics should be researched.
```
Comment (sims1253) [inline data/projects/implement_metrics_for_coala.md]: Again >.>

This is not a milestone that can be checked. The milestone should describe the desired state at the end of the period. 
And it has to be testable. Testing if something is researched is not really easy. You can have the student to propose a list that includes reasoning. But than the milestone is the existence of said list.

Comment (dracarys09) [inline data/projects/implement_metrics_for_coala.md]: yeah I know... I just addressed @SanketDG comment... changing it in a moment

File: data/projects/implement_metrics_for_coala.md [hunk 16]
```diff
@@ -1,34 +1,46 @@
...
+##### CODING PHASE 1
 
-* The API for bears to yield metrics has been implemented.
-* The API is properly documented and fully tested.
-* At least one Bear implementing the API exists.
+* The API for bears to yield metrics is implemented.
+* At least one Bear is implemented which uses this API.
 * A working (not clean or merged!) prototype of the CLI visualization
-exists.
+  exists.
```
Comment (sims1253) [inline data/projects/implement_metrics_for_coala.md]: why the whitespace?

Comment (dracarys09) [inline data/projects/implement_metrics_for_coala.md]: to align it with the upper line

File: data/projects/implement_metrics_for_coala.md (hunk 13 above)
Comment (SanketDG) [inline data/projects/implement_metrics_for_coala.md]: sorry ignore my previous comment

Comment (dracarys09) [inline data/projects/implement_metrics_for_coala.md]: np

File: data/projects/implement_metrics_for_coala.md (hunk 16 above)
Comment (sims1253) [inline data/projects/implement_metrics_for_coala.md]: aaaah
doing 2 windows next to each other made it seem strange

**Comment by sims1253** [comment]
ack 015edde07301d4117c452f3e20553822fe7a0958

**Comment by KVGarg** [comment]
[Radon ](https://radon.readthedocs.io/en/latest/index.html)also provide software metrics analysis on projects, then can we integrate that thing also with this project? or we need to implement software metrics that no other library has yet implemented?
